File: src/main/java/aristata/programmers/d20190316/SumOfDivisors.java
package aristata.programmers.d20190316;

public class SumOfDivisors {
    public static void main(String[] args) {
        SumOfDivisors a = new SumOfDivisors();
        int result = a.solution(12);
        System.out.println();
        System.out.println(result);
    }

    private int solution(int num) {
        int answer = 0;
        for (int i = 1; i <= num; i++) {
            if (num % i == 0)
                System.out.print(i + " ");
            answer += i;
        }
        return answer;
    }
}

File: src/main/kotlin/aristata/programmers/d20190316/SumOfDivisors.kt
package aristata.programmers.d20190316

class SumOfDivisors {

    fun solution(num: Int): Int {
        var answer = 0
        for (i in 1..num) {
            if (num % i == 0)
                print("$i ")
            answer += i
        }
        return answer
    }
}

fun main(args: Array<String>) {
    val a = SumOfDivisors()
    val result = a.solution(12)
    println()
    println(result)
}
